altos-rust/altos-core/libs/heap/free_list_allocator/src/lib.rs

```diff
@@ -65,9 +65,9 @@ pub extern fn __rust_allocate(size: usize, align: usize) -> *mut u8 {
 
 #[no_mangle]
 #[cfg(not(test))]
-pub extern fn __rust_deallocate(_ptr: *mut u8, _size: usize, _align: usize) {
+pub extern fn __rust_deallocate(ptr: *mut u8, size: usize, align: usize) {
```

Comment:
These functions are part of the required API for custom allocators (https://doc.rust-lang.org/book/custom-allocators.html). It would probably be better to make the argument names match the API exactly. For deallocate, it looks like it's actually `__rust_deallocate(ptr: *mut u8, _old_size: usize, _align: usize)`. We should check the other API functions as well.